api/envoy/data/accesslog/v3/accesslog.proto

```diff
@@ -215,6 +215,38 @@ message AccessLogCommon {
   // :ref:`stream_id <envoy_v3_api_field_data.accesslog.v3.AccessLogCommon.stream_id>` to
   // correlate all these intermediate log entries and final log entry.
   bool intermediate_log_entry = 27;
+
+  // For HTTP: Total number of bytes sent to the downstream by the http stream.
+  // For TCP: Total number of bytes sent to the downstream by the tcp proxy.
+  uint64 downstream_wire_bytes_sent = 28;
+
+  // For HTTP: Total number of bytes received from the downstream by the http stream. Envoy over counts sizes of received HTTP/1.1 pipelined requests by adding up bytes of requests in the pipeline to the one currently being processed.
+  // For TCP: Total number of bytes received from the downstream by the tcp proxy.
+  uint64 downstream_wire_bytes_received = 29;
+
+  // For HTTP: Number of header bytes sent to the downstream by the http stream.
+  // For TCP: Not implemented (0).
+  uint64 downstream_header_bytes_sent = 30;
+
+  // For HTTP: Number of header bytes received from the downstream by the http stream.
+  // For TCP: Not implemented (0).
+  uint64 downstream_header_bytes_received = 31;
```

Comment:
Is there a good position for these fields? Seems that these fields are only used by HTTP protocol and IMO they shouldn't added to `AccessLogCommon`? Or the `header` could also be header of other protocol (e.g. thrift)?

Reply:
Yeah, good catch, I think these belong in the respective parent message for HTTP.

Reply:
Thanks for taking a look!

IIUC, you're saying that the header specific fields should be moved inside of the HTTP request/response properties fields. If I understood correctly, does the following sketch of where the new fields should be instead look correct?

```
AccessLogCommon
	- downstream_wire_bytes_sent
	- downstream_wire_bytes_received
	- upstream_wire_bytes_sent
	- upstream_wire_bytes_received

HTTPAccessLogEntry.HTTPRequestProperties
	- downstream_header_bytes_received
	- upstream_header_bytes_sent

HTTPAccessLogEntry.HTTPResponseProperties
	- upstream_header_bytes_received
	- downstream_header_bytes_sent
```